Develop

LuckyFay12 opened the pull request.

pascalplata committed:
[add-reviews] add reviews

- refactor FilmDbStorage
- add storage, service and controller for Reviews
- add table to schema.sql
- minor additions and updates

pascalplata committed:
[add-reviews] checkstyle fix

pascalplata committed:
[add-reviews] delete commented lines

pascalplata committed:
[add-reviews] delete commented lines and add comments to code

LuckyFay12 committed:
Merge pull request #5 from LuckyFay12/add-reviews

[add-reviews] add reviews

LuckyFay12 committed:
feat: add directors functionality

LuckyFay12 committed:
Merge branch 'develop' into add-director

# Conflicts:
#	src/main/resources/schema.sql

PascalABCD committed:
Merge pull request #7 from LuckyFay12/add-director

Add director

LuckyFay12 committed:
add search functionality

LuckyFay12 committed:
add search functionality

LuckyFay12 committed:
Rename method getFilmsBySearch to getFilmsByParam

LuckyFay12 committed:
Merge pull request #8 from LuckyFay12/add-search

add search functionality

pascalplata committed:
[add-feed] add user feed

- create events table
- add feed storage, controller and service
- add feed events to userservice, filmservice, reviewservice

PascalABCD committed:
Merge pull request #9 from LuckyFay12/add-feed

[add-feed] add user feed

LuckyFay12 committed:
Add filtered popular films

LuckyFay12 committed:
Merge pull request #10 from LuckyFay12/add-most-populars

Add filtered popular films

pascalplata committed:
[add-remove-endpoint] remove films and users by id

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

LuckyFay12 committed:
Merge pull request #14 from LuckyFay12/add-remove-endpoint

[add-remove-endpoint] remove films and users by id

LuckyFay12 committed:
Add common films

LuckyFay12 committed:
Add common films

LuckyFay12 committed:
Add common films

PascalABCD committed:
Merge pull request #18 from LuckyFay12/add-common-films

Add common films

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Merge branch 'develop' into add-recommendations

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Merge branch 'add-recommendations' of https://github.com/LuckyFay12/java-filmorate into add-recommendations

# Conflicts:
#	src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java
#	src/main/java/ru/yandex/practicum/filmorate/service/UserService.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmDbStorage.java
#	src/main/java/ru/yandex/practicum/filmorate/storage/film/FilmStorage.java

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

Dead-Rey committed:
Добавление рекомендаций

pascalplata committed:
[fix-tests] test fixes

- check userId in getUserEvents
- use MERGE instead of INSERT for likes
- check userId and filmId for likes
- minor fix for ReviewService

LuckyFay12 committed:
Merge pull request #19 from LuckyFay12/add-recommendations

add-recommendations

LuckyFay12 committed:
Merge pull request #20 from LuckyFay12/fix-tests

[fix-tests] test fixes

LuckyFay12 committed:
Update popular films

pascalplata committed:
[develop] clean up schema.sql

LuckyFay12 committed:
Update schema.sql

VadimZharkov commented on src/main/java/ru/yandex/practicum/filmorate/handlers/GlobalExceptionHandler.java:
```diff
@@ -29,6 +28,18 @@ public ApiError handleUserNotFound(UserNotFoundException e) {
 
+    @ExceptionHandler(DirectorNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public Map<String, Object> handleDirectorNotFound(DirectorNotFoundException e) {
```
Здесь для единообразия тоже лучше ApiError использовать

VadimZharkov commented on src/main/java/ru/yandex/practicum/filmorate/exception/NotFoundException.java:
```diff
@@ -0,0 +1,7 @@
+package ru.yandex.practicum.filmorate.exception;
+
+public class NotFoundException extends RuntimeException {
```
Нигде не используется

VadimZharkov commented on src/main/java/ru/yandex/practicum/filmorate/storage/event/EventDbStorage.java:
```diff
@@ -0,0 +1,59 @@
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        // если timestamp не указан, используем текущее время в миллисекундах
+        long timestamp = event.getTimestamp() != null ? event.getTimestamp() : System.currentTimeMillis();
```
Данная проверка - это бизнес-правило, поэтому лучше вынести в класс-сервис

VadimZharkov reviewed (changes requested):
Поздравляю с первой групповой работой. Хорошая структура проекта и схема БД.  Вы отлично поработали. Хочу отметить, чтобы закрепилось, классы-хранилища должны быть достаточно примитивные, их задача просто хранить все что пришло от класса-сервиса. Все проверки связанные с консистентностью данных и бизнес-логикой - это зона ответственности классов-сервисов.  Со временем вы будете лучше чувствовать зоны ответственности разных слоев приложения. По самой работе оставил несколько комментариев.

pascalplata committed:
[review-fixes] review fixes

- use ApiError for directors not found
- move timestamp logic to service layer
- remove NotFoundException.java

LuckyFay12 committed:
Merge pull request #22 from LuckyFay12/review-fixes

Review fixes
